WordPress/Classes/Services/Stories/StoryEditor.swift

```diff
@@ -112,16 +128,60 @@ class StoryEditor: CameraController {
                  tagCollection: nil,
                  saveDirectory: saveDirectory)
 
-        cameraHandler = CameraHandler(created: { [weak self] _ in
+        cameraHandler = CameraHandler(created: { [weak self] media in
+            self?.poster = StoryPoster(context: post.blog.managedObjectContext ?? ContextManager.shared.mainContext, mediaFiles: mediaFiles)
+            let postMedia: [StoryPoster.MediaItem] = media.compactMap { result in
+                switch result {
+                case .success(let item):
+                    guard let item = item else { return nil }
+                    return StoryPoster.MediaItem(url: item.output, size: item.size, archive: item.archive, original: item.unmodified)
+                case .failure:
+                    return nil
+                }
+            }
+
+            guard let self = self else { return }
+            let uploads: (String, [Media])? = self.poster?.upload(mediaItems: postMedia, post: post as! Post, completion: { post in
+                uploaded(post)
+            })
+
+            if let firstMediaFile = mediaFiles?.first {
+                let processor = GutenbergBlockProcessor(for: "wp:jetpack/story", replacer: { block in
+                    let mediaFiles = block.attributes["mediaFiles"] as? [[String: Any]]
+                    if let mediaFile = mediaFiles?.first, mediaFile["url"] as? String == firstMediaFile.url {
+                        return uploads?.0
+                    } else {
+                        return nil
+                    }
+                })
+                post.content = processor.process(post.content ?? "")
+            } else {
+                post.content = uploads?.0
+            }
+
+            try! post.managedObjectContext?.save()
+
+            updated(.success(post as! Post))
+
             if publishOnCompletion {
-                self?.publishPost(action: .publish, dismissWhenDone: true, analyticsStat:
+                self.publishPost(action: .publish, dismissWhenDone: true, analyticsStat:
                                     .editorPublishedPost)
             } else {
-                self?.dismiss(animated: true, completion: nil)
+                self.dismiss(animated: true, completion: nil)
             }
         })
         self.delegate = cameraHandler
     }
+
+    func populate(with files: [MediaFile], completion: @escaping (Result<Void, Error>) -> Void) {
+        storyLoader?.download(files: files, for: post) { [weak self] output in
+            DispatchQueue.main.async {
+                self?.show(media: output)
+                completion(.success(()))
+                print(output)
```

Comment:
Nit: Do you need this print or DDLog can be used instead?

Reply:
I'll remove this and add some more proper logging in the future.